Loop/Managers/RemoteDataManager.swift

```diff
@@ -8,34 +8,177 @@
+                let insulinOnBoard: InsulinValue?
+
+                switch result {
+                case .success(let value):
+                    insulinOnBoard = value
+                case .failure(let error):
+                    insulinOnBoard = nil
+
+                    if loopError == nil {
+                        loopError = error
+                    }
+                }
+
+                self.uploadLoopStatus(
+                    insulinOnBoard: insulinOnBoard,
+                    carbsOnBoard: carbsOnBoard,
+                    predictedGlucose: predictedGlucose,
+                    recommendedTempBasal: recommendedTempBasal,
+                    recommendedBolus: recommendedBolus,
+                    loopError: loopError
+                )
+
+                self.uploadSettings()
+            }
+        }
+    }
+
+    func uploadLoopStatus(
+        insulinOnBoard: InsulinValue? = nil,
+        carbsOnBoard: CarbValue? = nil,
+        predictedGlucose: [GlucoseValue]? = nil,
+        recommendedTempBasal: (recommendation: TempBasalRecommendation, date: Date)? = nil,
+        recommendedBolus: Double? = nil,
+        lastReservoirValue: ReservoirValue? = nil,
+        pumpManagerStatus: PumpManagerStatus? = nil,
+        glucoseTargetRangeSchedule: GlucoseRangeSchedule? = nil,
+        scheduleOverride: TemporaryScheduleOverride? = nil,
+        glucoseTargetRangeScheduleApplyingOverrideIfActive: GlucoseRangeSchedule? = nil,
+        loopError: Error? = nil)
+    {
+        remoteData.forEach {
+            $0.uploadLoopStatus(
+                insulinOnBoard: insulinOnBoard,
+                carbsOnBoard: carbsOnBoard,
+                predictedGlucose: predictedGlucose,
+                recommendedTempBasal: recommendedTempBasal,
+                recommendedBolus: recommendedBolus,
+                lastReservoirValue: lastReservoirValue ?? deviceDataManager.loopManager.doseStore.lastReservoirValue,
+                pumpManagerStatus: pumpManagerStatus ?? deviceDataManager.pumpManagerStatus,
+                glucoseTargetRangeSchedule: glucoseTargetRangeSchedule,
+                scheduleOverride: scheduleOverride,
+                glucoseTargetRangeScheduleApplyingOverrideIfActive: glucoseTargetRangeScheduleApplyingOverrideIfActive,
+                loopError: loopError)
         }
     }
 
-    private let keychain = KeychainManager()
+    func upload(glucoseValues values: [GlucoseValue], sensorState: SensorDisplayable?) {
+        remoteData.forEach { $0.upload(glucoseValues: values, sensorState: sensorState) }
+    }
+
+    func upload(pumpEvents events: [PersistedPumpEvent], fromSource source: String, completion: @escaping (Result<[URL], Error>) -> Void) {
+        // TODO: How to handle completion correctly
+        if remoteData.count > 0 {
+            remoteData[0].upload(pumpEvents: events, fromSource: source, completion: completion)
+        }
```

Comment:
Why only the first element? Similar comment for code below.

Reply:
This is a placeholder that will only work for one `RemoteData` service at a time. However, at this point with the code there is only one `RemoteData` service anyways, namely `NightscoutService`. 

This is one reason why we are moving from the current "push" model to a "pull" model. The "pull" model will allow multiple `RemoteData` services to be active at the same time (eg. `NightscoutService` and `TidepoolService`).

So, this is a "good enough for now" situation that will be updated with the next round of updates to the `RemoteData` service and `NightscoutService`.